pkg/types/azure/disk.go

```diff
@@ -0,0 +1,65 @@
+package azure
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	RxDiskEncryptionSetID   = regexp.MustCompile(`(?i)^/subscriptions/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/resourceGroups/([-a-z0-9_().]{0,89}[-a-z0-9_()])/providers/Microsoft\.Compute/diskEncryptionSets/([-a-z0-9_]{1,80})$`)
+	RxSubscriptionID        = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+	RxResourceGroup         = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+	RxDiskEncryptionSetName = regexp.MustCompile(`^[-a-z0-9_]{1,80}$`)
+)
+
+// DiskEncryptionSet defines the configuration for a disk encryption set.
+type DiskEncryptionSet struct {
```

Comment:
As `DiskEncryptionSet` is used as a field in `OSDisk` I think it would be more readable to have `OSDisk` appear first  and `DiskEncryptionSet` would be defined below.

Reply:
Fixed.